src/renderers/dom/client/utils/DOMLazyTree.js

```diff
@@ -53,8 +53,16 @@ function insertTreeChildren(tree) {
 
 var insertTreeBefore = createMicrosoftUnsafeLocalFunction(
   function(parentNode, tree, referenceNode) {
-    parentNode.insertBefore(tree.node, referenceNode);
-    insertTreeChildren(tree);
+    // Document Fragments in IE11, Edge (and possibly others) won't update
```

Comment:
Sorry if it was unclear: this is how doc fragments work in all browsers. But we insert things in a different order in IE11 and Edge which is why they exhibit this bug.